README.md

```diff
@@ -1,317 +1,177 @@
+| `.registerStreamHandler`  | Add a event handler for a `Future`  |
 
-With this mixin you can register handlers for `Streams`, `ValueListenables` and Futures, and the mixin will dispose everything for you as soon as the widget gets destroyed.
-
-```Dart
-class TestStateLessWidget1 extends StatelessWidget with GetItMixin {
+The first param in the `register` methods is a `select` delegate that can be used to bind to a specific field. The second param is the handler delegate itself:
+```dart
+class MyWidget extends StatelessWidget with GetItMixin {
   @override
   Widget build(BuildContext context) {
-    /// Registers a handler for a valueListenable
-    registerHandler((Model x) => x.name, (context,name,_) 
-        => showNameDialog(context,name));
-        
-    registerStreamHandler((Model x) => x.userNameUpdates, (context,name,_) 
-        => showNameDialog(context,name));
-
-    registerFutureHandler((Model x) => x.initializationReady, (context,__,_) 
-        => Navigator.of(context).push(....));
-    return Column(
-      children: [
-        //...whatever widgets needed 
-      ],
-    );
+    registerHandler(
+        (Model x) => x.name,
+        (context, value, cancel) => showNameDialog(context, value));
+    ...
   }
 }
 ```
-For instance you could register a handler for `thrownExceptions` of a `flutter_command` while you use `watch()` to get the values.
 
-In the example above you see that the handler function has a third parameter that we ignored. Your handler gets a dispose function passed there that a handler could use to kill a registration from within itself.
+In the example above you see that the handler function receives the value that is returned from the select delegate (`(Model x) => x.name`), as well as a `cancel` function that the handler can call to cancel registration at any time.
 
-### allReady() & isReady()
-If you already used the synchronization functions from GetIt you know both of this functions (otherwise check them out in the GetIt readme). The mixin variant returns the actual status as `bool` value and trigger a rebuild when this status changes. Additionally you can register handlers that are called when the status is `true`.
+As with `watch` calls, all registered handlers are cleaned up when the widget is destroyed!
 
-```Dart
-class TestStateLessWidget1 extends StatelessWidget with GetItMixin {
-  @override
-  Widget build(BuildContext context) {
-    final isReady = allReady();
+# Rules
 
-    if (isReady) {
-      return MyMainPageContent();
-    } else {
-      return CircularProgressIndicator();
-    }
-}
-```
-or with the handler:
+There are some important rules to follow in order to avoid bugs with the `watch` methods:
+* `watch` methods should only be called within `build()`
+* must be called on every build, in the same order (no conditional watching)
+* do not use them inside of a builder as it will break the mixins ability to rebuild
 
```

Comment:
Ya I was going to ask you about this cause I don't know... it's not mentioned in the current docs. Seems like it would be a nice use case...registering each tick seems odd to me if you're in a stateful widget already.